tools/ci_build/github/azure-pipelines/templates/win-cpu-ci.yml

```diff
@@ -84,21 +84,20 @@ jobs:
     workingDirectory: '$(Build.SourcesDirectory)'
     displayName: 'API Documentation Check and generate'
 
-  - script: |
-     python -m pip install -q setuptools wheel numpy
-    workingDirectory: '$(Build.BinariesDirectory)'
-    displayName: 'Install python modules'
 
-  - ${{ if eq(parameters.RunOnnxRuntimeTests, true) }}:
-      - powershell: |
-         $Env:USE_MSVC_STATIC_RUNTIME=1
-         $Env:ONNX_ML=1
-         $Env:CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=OFF -DProtobuf_USE_STATIC_LIBS=ON -DONNX_USE_LITE_PROTO=ON -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET=${{ parameters.buildArch }}-windows-static"
-         python setup.py bdist_wheel
-         python -m pip uninstall -y onnx -qq
-         Get-ChildItem -Path dist/*.whl | foreach {pip --disable-pip-version-check install --upgrade $_.fullname}
-        workingDirectory: '$(Build.SourcesDirectory)\cmake\external\onnx'
-        displayName: 'Install ONNX'
+  - powershell: |
+     $Env:USE_MSVC_STATIC_RUNTIME=1
+     $Env:ONNX_ML=1
+     $Env:CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=OFF -DProtobuf_USE_STATIC_LIBS=ON -DONNX_USE_LITE_PROTO=ON -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET=${{ parameters.buildArch }}-windows-static"
+     python -m pip uninstall -y ort-nightly-gpu ort-nightly onnxruntime onnxruntime-gpu onnxruntime-training onnxruntime-directml ort-nightly-directml onnx -qq
+     python -m pip install -r $(Build.SourcesDirectory)\tools\ci_build\github\windows\${{ parameters.buildArch }}\requirements.txt
+    displayName: 'Install ONNX'
+  - ${{ if eq(parameters.buildArch, 'x64') }}:
+    - task: PythonScript@0
+      displayName: 'Generate Test Data'
+      inputs:
+        scriptPath: '$(Build.SourcesDirectory)\onnxruntime\test\python\generate_test_data.py'
+        arguments: generate-data -o $(Build.BinariesDirectory)\${{ parameters.BuildConfig }}\${{ parameters.BuildConfig }}
```

Comment:
Can we generate the models and check them in so they are run when you do local testing?

Reply:
Yes.

What I really want is for it to be simple to run these by default locally when the other python tests run.

Right now it seems like I need to be aware that this script exists and needs to be manually run. Without that knowledge the first place you could see one of these tests fail is in the CI. You might try and run the same command locally and it won't fail, because you also need to be aware of the step in the CI where this extra test data was generated. 

Given there's already a testdata folder under test\python, and the models should be small, it seems like less potential for wasted developer time if we just check in the results of the model generation.

Reply:
Then it will be very like what ONNX has. I will check-in the python code and models, but do not run the python code in our pipelines.  We will have the same problem as ONNX has:

1. The script and the test data could often get out of sync. ONNX still has this headache. 
2. Sometimes, you will need to update thousands of files for a very small code change. For example, ONNX made a patch release to update over 2000 files which was just for fixing a tiny problem in a single op.

TF2ONNX is small. TF is the biggest dependency here. And we already have it in our pipelines.  In my experience, installing TF2ONNX only takes a few seconds and never gave me any problem if TF is already there.